butte-build/src/ir/transform.rs

```diff
@@ -313,6 +310,53 @@ impl<'a> Builder<'a> {
         })
     }
 
+    fn new_rpc(&mut self, rpc: &ast::Rpc<'a>, pos: usize) -> Result<bool> {
+        let ident = &self.make_fully_qualified_ident(ir::Ident::from(&rpc.id));
+
+        let mut methods = Vec::new();
+
+        for m in &rpc.methods {
+            let ident = ir::Ident::from(&m.id);
+            let snake_ident = ir::Ident::from(m.id.as_ref().to_snake_case());
+            let request_type = ir::DottedIdent::from(&m.request_type);
+            let response_type = ir::DottedIdent::from(&m.response_type);
+            let streaming = match &m.metadata {
+                Some(m) => match m.values.get(&ast::Ident::from("streaming")) {
+                    Some(Some(ast::Single::String("client"))) => ir::RpcStreaming::Client,
+                    Some(Some(ast::Single::String("server"))) => ir::RpcStreaming::Server,
+                    Some(Some(ast::Single::String("bidi"))) => ir::RpcStreaming::Bidi,
+                    None => ir::RpcStreaming::None,
+                    Some(Some(s)) => return Err(anyhow!("Unknown streaming type: {:?}", s)),
+                    Some(None) => {
+                        return Err(anyhow!(
+                        "Missing streaming type: possible values are `client`, `server` and `bidi`"
+                    ))
+                    }
```

Comment:
Formatting here looks a bit off. Should the two previous lines be indented?

Reply:
I'll have a look, but this is likely `rustfmt`'s doing.

Reply:
Cool, I wouldn't worry about it too much.